src/components/Channel/Channel.js

```diff
@@ -1,148 +1,674 @@
+    }
+  };
+
+  const markReadThrottled = throttle(markRead, 500, {
+    leading: true,
+    trailing: true,
+  });
+
+  const copyChannelState = () => {
+    if (unmounted) return;
+    setLoading(false);
+    setMembers(channel.state.members);
+    setMessages(channel.state.messages);
+    setRead(channel.state.read);
+    setTyping(Immutable({}));
+    setWatcherCount(channel.state.watcher_count);
+    setWatchers(channel.state.watchers);
+
+    if (channel.countUnread() > 0) {
+      markReadThrottled();
+    }
+  };
+
+  const addToEventHistory = (e) => {
+    const lastMessageId = messages.length
+      ? messages[messages.length - 1].id
+      : 'none';
+
+    setEventHistory((prevState) => {
+      if (!prevState[lastMessageId]) {
+        return { ...prevState, [lastMessageId]: [e] };
+      } else {
+        return {
+          ...prevState,
+          [lastMessageId]: [...prevState[lastMessageId], e],
+        };
+      }
+    });
+  };
+
+  const handleEventStateChange = (channelState) => {
+    setMessages(channelState.messages);
+    setRead(channelState.read);
+    setTyping(channelState.typing);
+    setWatcherCount(channelState.watcher_count);
+    setWatchers(channelState.watchers);
+  };
+
+  const handleEventStateThrottled = throttle(handleEventStateChange, 500, {
+    leading: true,
+    trailing: true,
+  });
+
+  const handleEvent = (e) => {
+    if (unmounted) return;
+
+    if (thread) {
+      const updatedThreadMessages =
+        channel.state.threads[thread.id] || threadMessages;
+      setThreadMessages(updatedThreadMessages);
+    }
+
+    if (thread && e.message?.id === thread.id) {
+      const updatedThread = channel.state.messageToImmutable(e.message);
+      setThread(updatedThread);
+    }
+
+    if (e.type === 'member.added') addToEventHistory(e);
+    if (e.type === 'member.removed') addToEventHistory(e);
+
+    handleEventStateThrottled(channel.state);
+  };
+
+  const listenToChanges = () => {
+    // The more complex sync logic is done in Chat.js
+    // listen to client.connection.recovered and all channel events
+    client.on('connection.recovered', handleEvent);
+    channel.on(handleEvent);
+  };
+
+  const initChannel = async () => {
+    let initError = false;
+
+    if (!channel.initialized) {
+      try {
+        await channel.watch();
+      } catch (e) {
+        if (unmounted) return;
+        setError(e);
+        setLoading(false);
+        initError = true;
+      }
+    }
+
+    setLastRead(new Date());
+    if (!initError) {
+      copyChannelState();
+      listenToChanges();
+    }
+  };
+
+  /**
+   * MESSAGE METHODS
+   */
+
+  const updateMessage = (updatedMessage, extraState = {}) => {
+    channel.state.addMessageSorted(updatedMessage);
+
+    if (thread && updatedMessage.parent_id) {
+      extraState.threadMessages =
+        channel.state.threads[updatedMessage.parent_id] || [];
+      setThreadMessages(extraState.threadMessages);
+    }
+
+    if (unmounted) return;
```

Comment:
what's the point of doing all the setThreadMessages stuff above this in this function if we're unmounted?

Reply:
similar to the other comment, not sure, just keeping the order of things as they were